common/ast/src/ast/statement.rs

```diff
@@ -1068,50 +1072,49 @@ impl<'a> Display for Statement<'a> {
-                                .iter()
-                                .map(|p| p.to_string())
-                                .collect::<Vec<_>>()
-                                .join(", ")
-                        )?;
+                    AccountMgrSource::Role { role } => write!(f, " ROLE {role}")?,
+                    AccountMgrSource::Privs { privileges, level } => {
+                        write!(f, " ")?;
+                        write_comma_separated_list(f, privileges.iter().map(|p| p.to_string()))?;
                         write!(f, " ON")?;
                         match level {
-                            GrantLevel::Global => write!(f, " *.*")?,
-                            GrantLevel::Database(database_name) => {
+                            AccountMgrLevel::Global => write!(f, " *.*")?,
+                            AccountMgrLevel::Database(database_name) => {
                                 if let Some(database_name) = database_name {
                                     write!(f, " {database_name}.*")?;
+                                } else {
+                                    write!(f, " *")?;
                                 }
                             }
-                            GrantLevel::Table(database_name, table_name) => {
+                            AccountMgrLevel::Table(database_name, table_name) => {
                                 if let Some(database_name) = database_name {
                                     write!(f, " {database_name}.{table_name}")?;
+                                } else {
+                                    write!(f, " {table_name}")?;
                                 }
                             }
                         }
                     }
-                    GrantSource::ALL { level, .. } => {
+                    AccountMgrSource::ALL { level, .. } => {
                         write!(f, " ALL PRIVILEGES")?;
                         write!(f, " ON")?;
                         match level {
-                            GrantLevel::Global => write!(f, " *.*")?,
-                            GrantLevel::Database(database_name) => {
+                            AccountMgrLevel::Global => write!(f, " *.*")?,
+                            AccountMgrLevel::Database(database_name) => {
                                 if let Some(database_name) = database_name {
                                     write!(f, " {database_name}.*")?;
                                 } else {
                                     write!(f, " *")?;
                                 }
                             }
-                            GrantLevel::Table(database_name, table_name) => {
+                            AccountMgrLevel::Table(database_name, table_name) => {
                                 if let Some(database_name) = database_name {
                                     write!(f, " {database_name}.{table_name}")?;
+                                } else {
```

Comment:
About this comment https://github.com/datafuselabs/databend/pull/6066#discussion_r901315699 already modified at there.@andylokandy